site/source/docs/porting/simd.rst

```diff
@@ -3,27 +3,76 @@
+Using SIMD with WebAssembly
+===========================
+
+Emscripten supports the `WebAssembly SIMD <https://github.com/webassembly/simd/>`_ feature. There are five different ways to leverage WebAssembly SIMD in your C/C++ programs:
+
+1. Enable LLVM/Clang SIMD autovectorizer to automatically target WebAssembly SIMD, without requiring changes to C/C++ source code.
+2. Write SIMD code using the GCC/Clang SIMD Vector Extensions (``__attribute__((vector_size(16)))``)
+3. Write SIMD code using the WebAssembly SIMD intrinsics (``#include <wasm_simd128.h>``)
+4. Compile existing SIMD code that uses the x86 SSE, SSE2, SSE3, SSSE3, SSE4.1, SSE4.2 or 128-bit subset of the AVX intrinsics (``#include <*mmintrin.h>``)
+5. Compile existing SIMD code that uses the ARM NEON intrinsics (``#include <arm_neon.h>``)
+
+These techniques can be freely combined in a single program.
+
+To enable any of the five types of SIMD above, pass the WebAssembly-specific ``-msimd128`` flag at compile time. This will also turn on LLVM's autovectorization passes. If that is not desirable, additionally pass flags ``-fno-vectorize -fno-slp-vectorize`` to disable the autovectorizer. See `Auto-Vectorization in LLVM <https://llvm.org/docs/Vectorizers.html>`_ for more information.
+
+WebAssembly SIMD is supported by
+
+* Chrome ≥ 91 (May 2021),
+
+* Firefox ≥ 89 (June 2021),
+
+* Safari ≥ 16.4 (March 2023) and
+
+* Node.js ≥ 16.4 (June 2021).
 
-Emscripten supports the `WebAssembly SIMD proposal <https://github.com/webassembly/simd/>`_ when using the WebAssembly LLVM backend. To enable SIMD, pass the -msimd128 flag at compile time. This will also turn on LLVM's autovectorization passes, so no source modifications are necessary to benefit from SIMD.
+See `WebAssembly Roadmap <https://webassembly.org/roadmap/>`_ for details about other VMs.
 
-At the source level, the GCC/Clang `SIMD Vector Extensions <https://gcc.gnu.org/onlinedocs/gcc/Vector-Extensions.html>`_ can be used and will be lowered to WebAssembly SIMD instructions where possible. In addition, there is a portable intrinsics header file that can be used.
+An upcoming `Relaxed SIMD proposal <https://github.com/WebAssembly/relaxed-simd/tree/main/proposals/relaxed-simd>`_ will add more SIMD instructions to WebAssembly.
+
+================================
+GCC/Clang SIMD Vector Extensions
+================================
+
+At the source level, the GCC/Clang `SIMD Vector Extensions <https://gcc.gnu.org/onlinedocs/gcc/Vector-Extensions.html>`_ can be used and will be lowered to WebAssembly SIMD instructions where possible.
+
+This enables developers to create custom wide vector types via typedefs, and use arithmetic operators (+,-,*,/) on the vectorized types, as well as allow individual lane access via the vector[i] notation. However, the `GCC vector built-in functions <https://gcc.gnu.org/onlinedocs/gcc/x86-Built-in-Functions.html>`_ are not available. Instead, use the WebAssembly SIMD Intrinsics functions below.
```

Comment:
Is it necessary to point out that the linked built-in functions are not available? They're documented as being for x86, so I don't think anyone should expect them to be available in the first place.

Reply:
I think it is helpful, because developers are already given "here's the native SSE and NEON intrinsics headers, go compile your existing code" - so someone might extrapolate from that: "well, I've got our GCC vectorized code, can I compile that?"